src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -248,29 +248,58 @@ public IObservable<Unit> UnmarkLocalBranch(ILocalRepositoryModel repository)
             });
         }
 
-        public IObservable<string> ExtractFile(ILocalRepositoryModel repository, string commitSha, string fileName)
+        public IObservable<string> ExtractFile(
+            ILocalRepositoryModel repository,
+            IModelService modelService,
+            string commitSha,
+            string fileName,
+            string fileSha)
         {
             return Observable.Defer(async () =>
             {
                 var repo = gitService.GetRepository(repository.LocalPath);
                 await gitClient.Fetch(repo, "origin");
-                var result = await gitClient.ExtractFile(repo, commitSha, fileName);
+                var result = await gitClient.ExtractFile(repo, commitSha, fileName) ??
+                             await modelService.GetFileContents(repository, commitSha, fileName, fileSha);
```

Comment:
If an exception is thrown inside an `Observable.Defer` then it will be converted into an `OnError` handler, so we're safe from crashing at least. The question is what do we want to do with any errors encountered? Show in the status bar? In the PR details pane?